### ISSUE TITLE
Hacker Cat 🐱‍💻

### ISSUE THREAD

**Issue by alexkadis** [hfg-gmuend/openmoji#95]
Hacker Cat is a [non-standard emoji sequence that was created by Microsoft](https://emojipedia.org/hacker-cat/). I thought it would be fun to include in OpenMoji.

Codepoints:
🐱 U+1F431
‍ U+200D
💻 U+1F4BB

This was designed using a combination of the OpenMoji  😾Pouting Cat Face and 💻 Personal Computer emojis.

PNG (128 x 128 px)
![hacker-cat-openmoji](https://user-images.githubusercontent.com/4106499/70351888-e7d53900-1837-11ea-87dc-22dd1283e036.png)

SVG (I can't seem to upload svgs as-is to Github)
[hacker-cat-openmoji.svg.txt](https://github.com/hfg-gmuend/openmoji/files/3933761/hacker-cat-openmoji.svg.txt)

**Comment by carlinmack** [comment]
I like it! I think it might be improved by bringing the laptop face on and the back having a trapezium shape. If it doesn't look like a laptop any more, maybe put a small circle in the middle to denote the logo on mac, surface, etc laptops

**Comment by b-g** [comment]
This is a fun one! :) But before we are able to include the Hacker Cat ... it needs a bit more ❤️! Please have a look at the https://openmoji.org/styleguide/ and consider Carlin's feedback. Also please submit the iterations here showing the grid (guide lines) layer and put it next to a few related ones eg. [woman technologist](https://openmoji.org/library/#search=tech&emoji=1F469-200D-1F4BB)

<img width="768" alt="Screen Shot 2019-12-09 at 09 03 57" src="https://user-images.githubusercontent.com/480224/70441707-e045a800-1a62-11ea-8826-f7ca44d3f135.png">

**Comment by alexkadis** [comment]
Thank you @carlinmack and @b-g for the feedback. How about this version?

Color:
![hacker-cat-512](https://user-images.githubusercontent.com/4106499/70453843-6ff55180-1a77-11ea-8bf6-ceb9ec816c4b.png)

On the grid:
![hacker-cat-grid](https://user-images.githubusercontent.com/4106499/70453842-6ff55180-1a77-11ea-99c5-a5d431bb3106.png)

On the grid, outline only:
![hacker-cat-grid-outline-only](https://user-images.githubusercontent.com/4106499/70454006-be0a5500-1a77-11ea-950e-61599f5fd8d8.png)

**Comment by carlinmack** [comment]
Nice looks good! We might discuss editing woman technologist to have the same laptop

**Comment by b-g** [comment]
Yes it is much better! However it is very unusual to use the entire y-space ... it will look very tall next to the technologist ones. Hence I would vote: 

- to make the cat (head) smaller to fit better in terms of size to technologist
- would leave the notebook as it is, even if different to other notebooks
- maybe: make the notebook logo in a lighter grey or full black
- maybe: add the notebook logo in the line layer (or line-supplement layer)
- maybe: add the fun headband like in the original emoji (but not in red, please no code kamikaze connotations)

![hacker-cat_1f431-200d-1f4bb](https://user-images.githubusercontent.com/480224/70461309-66271a80-1a86-11ea-80d8-9814c0d1a2ce.png)

**Comment by alexkadis** [comment]
@b-g how's this?
- I made the cat head roughly the same size as the woman technologist.
- I made the laptop logo black and moved it to the line-layer.
- I adapted the 🎀 ribbon emoji and 👔 necktie emojis to make the bandana.

With standard dimensions:
![hacker-cat](https://user-images.githubusercontent.com/4106499/70467156-a7252c00-1a92-11ea-89cf-9808281a17b3.png)

Zoomed in (my preference):
![hacker-cat-zoomed-in](https://user-images.githubusercontent.com/4106499/70467166-adb3a380-1a92-11ea-9c99-dde8365fedf0.png)

Grid:
![hacker-cat-grid](https://user-images.githubusercontent.com/4106499/70467347-071bd280-1a93-11ea-90d7-ebb1fea99062.png)

Outline:
![hacker-cat-grid-outline-only](https://user-images.githubusercontent.com/4106499/70467362-0c791d00-1a93-11ea-9b63-0cbe9f03609c.png)

**Comment by dnlutz** [comment]
I'm getting into the discussion a little late, but I have one remark:
the lines should have a constant line width of 2 px. At the moment they look uneven and it looks almost as if the cat is wearing eye shadow ; ) The line endings over the laptop also look a bit quirky, the should be rounded / circular.

I would not change the laptop of "woman technologist", IMO it's ok when the cat uses a different laptop.

**Comment by alexkadis** [comment]
@dnlutz I've made a new version that has 2px line widths only. I don't really like it much as it requires losing a **lot** of the detail. Anyone have suggestions?

Zoomed in:
![hacker-cat-2px-line-zoomed-in](https://user-images.githubusercontent.com/4106499/70563295-fb481300-1b5b-11ea-815d-e031b3885d18.png)

Standard 72px:
![hacker-cat-2px-line](https://user-images.githubusercontent.com/4106499/70563296-fbe0a980-1b5b-11ea-901c-9faa98dfefa5.png)

**Comment by carlinmack** [comment]
hmm have a look at the line widths of the original cat https://github.com/hfg-gmuend/openmoji/blob/master/src/smileys-emotion/cat-face/1F63E.svg in your editor and see what the they are. I think @dnlutz was wanting you to use the eyes from that emoji as they were somehow altered in the process to now

**Comment by alexkadis** [comment]
So that it works with the required 2px outlines, my options are to either make the face much larger than [woman technologist](https://openmoji.org/library/#search=tech&emoji=1F469-200D-1F4BB), making it take up much more room on vertical space than is proper, or drastically simplify the cat.

Latest version:
<img src="https://user-images.githubusercontent.com/4106499/70573655-6b609400-1b70-11ea-8c8c-f2c05ab9503a.png" width="72px" alt="hacker-cat"  />

Compared to...
<img src="https://openmoji.org/data/color/svg/1F469-200D-1F4BB.svg" width="72px" alt="woman-technologist"  />

Zoomed in:
<img src="https://user-images.githubusercontent.com/4106499/70573655-6b609400-1b70-11ea-8c8c-f2c05ab9503a.png" width="512px" alt="hacker-cat"  />


Grid lines:
![hacker-cat-grid-outline-only](https://user-images.githubusercontent.com/4106499/70573889-d611cf80-1b70-11ea-9e58-8a5e533bac1c.png)

Grid color and lines:
![hacker-cat-grid](https://user-images.githubusercontent.com/4106499/70573881-d27e4880-1b70-11ea-926b-b924df9b2a53.png)

**Comment by dnlutz** [comment]
A lot of important design principles are addressed in this thread. Therefore I will try to look at some of the issues in detail. Unfortunately, I won't get to it until the weekend at the earliest. But I'll take care of it asap ; )

**Comment by alexkadis** [comment]
Another version that's a bit clearer at smaller sizes.

Version above (Let's call it Version A):
<img src="https://user-images.githubusercontent.com/4106499/70573655-6b609400-1b70-11ea-8c8c-f2c05ab9503a.png" alt="Hacker Cat - Version A - at 16px" width="16px" /> <img src="https://user-images.githubusercontent.com/4106499/70573655-6b609400-1b70-11ea-8c8c-f2c05ab9503a.png" alt="Hacker Cat - Version A - at 32px" width="32px" /> <img src="https://user-images.githubusercontent.com/4106499/70573655-6b609400-1b70-11ea-8c8c-f2c05ab9503a.png" alt="Hacker Cat - Version A - at 72px" width="72px" />

New version (Let's call it Version B):
<img src="https://user-images.githubusercontent.com/4106499/70741500-6e32c480-1ce9-11ea-8d68-fa6b4e64bbc5.png" alt="Hacker Cat - Version B - at 16px" width="16px" /> <img src="https://user-images.githubusercontent.com/4106499/70741500-6e32c480-1ce9-11ea-8d68-fa6b4e64bbc5.png" alt="Hacker Cat - Version B - at 32px" width="32px" /> <img src="https://user-images.githubusercontent.com/4106499/70741500-6e32c480-1ce9-11ea-8d68-fa6b4e64bbc5.png" alt="Hacker Cat - Version B - at 72px" width="72px" />
![hacker-cat-simple](https://user-images.githubusercontent.com/4106499/70741500-6e32c480-1ce9-11ea-8d68-fa6b4e64bbc5.png)
![hacker-cat-simple-grid](https://user-images.githubusercontent.com/4106499/70741499-6d9a2e00-1ce9-11ea-8ca6-4a8ffda0bbe7.png)
![hacker-cat-simple-outline](https://user-images.githubusercontent.com/4106499/70741501-6e32c480-1ce9-11ea-833f-c6c9fea1512e.png)

**Comment by dnlutz** [comment]
As promised – some thoughts on some basic topics in this thread:

**Simplicity is not a bug – it's our key feature**

Compared to letters, Emoji are far too detailed and intricate. This makes them difficult to read and use. That's the reason why we developed OpenMoji as a simpler and more straightforward system. We don't want to be neither retro nor arrogant, but we consider current mainstream emoji design to be a historical aberration. 

<img width="1091" alt="emoji_history" src="https://user-images.githubusercontent.com/38292227/70903764-de9a5800-1fff-11ea-8bb8-e89c9090561c.png">

Graphical signs have to be easily identifiable and differentiable. They should also harmonize as well as possible with the surrounding text. That's why we stick to Albert Einstein's quote: "Everything should be made as simple as possible, but no simpler".

<img width="465" alt="emoji_text" src="https://user-images.githubusercontent.com/38292227/70903804-fd98ea00-1fff-11ea-8fdd-1f0ec234c9b1.png">

**Comment by dnlutz** [comment]
**Details**

Of course people enjoy beautiful details. Not only Charles Eames has recognized their importance: "The details are not the details. They make the design." However, simplicity is our central feature. We can never reproduce all the details of the world - so we shouldn't even try it. That's why we want all emoji to be as plain and simple as possible: from "Woman Mountain Biking: Medium-Dark Skin Tone" to all the flags.

Our long-term goal: as few anchor points as possible ;) That's why we want to further simplify all unnecessary details in future releases. 

![woman_mountainbike](https://user-images.githubusercontent.com/38292227/70903904-3d5fd180-2000-11ea-9b05-34ba7879e959.png)

![spain](https://user-images.githubusercontent.com/38292227/70903908-3fc22b80-2000-11ea-894e-eb5a09300920.png)

BTW "man and woman mountain biking" are interchanged – we have to fix that

**Comment by dnlutz** [comment]
**Line width**

A uniform line width of all our emoji is a central parameter of our design system. The default line width is 2 px. To ensure a uniform visual perception, the line width can be reduced to 1.6 px for small elements. If the line width is left at 2 px, small elements appear too thick compared to larger shapes. This optical effect is related to the interior shape:

![export](https://user-images.githubusercontent.com/38292227/70904026-9e87a500-2000-11ea-94e1-44374fd9cca7.png)

**Comment by dnlutz** [comment]
**Interior shapes**

"The white inner space of the letter is involved in the shape of the letter. While drawing, the type designer must constantly weigh the form against the counter form." Emil Ruder

Just as with type design, it is essential to keep an eye not only on the black outlines, but also on the negative forms they create. Too small and delicate interiors should be avoided, therefore the shape of the outlines must be changed if necessary. 

Small interior shapes often occur when double lines are used. Therefore double lines should be avoided if possible and replaced by single lines. If double lines are used, the minimum distance between them should be respected.

![export2](https://user-images.githubusercontent.com/38292227/70904094-d68ee800-2000-11ea-9488-1dda6a86a35b.png)
![export3](https://user-images.githubusercontent.com/38292227/70904097-d858ab80-2000-11ea-94b8-4f5bf1748f21.png)
![export4](https://user-images.githubusercontent.com/38292227/70904104-db539c00-2000-11ea-9e1c-f5d68c2c45a1.png)
![export5](https://user-images.githubusercontent.com/38292227/70904112-ddb5f600-2000-11ea-9e79-0b2f2f70bc14.png)

**Comment by dnlutz** [comment]
Back to Hackercat: the only thing i would recommend to take a closer look is the headband – the rest is fine for me. Maybe it could go in this direction?
<img width="310" alt="hackercat" src="https://user-images.githubusercontent.com/38292227/70904229-21a8fb00-2001-11ea-85d5-6c54b3ed60ed.png">

**Comment by dnlutz** [comment]
**Sizes**

Unified sizes are a tough topic for emoji. Our first aid is the base square. From this base square we have derived basic sizes for other basic geometric shapes. Nevertheless, a visual check after the design is indispensable. We should present the basic sizes for different emoji categories in more detail in a guidelines update.

That was a lot of text in the Hackercat thread now. But since we have to extend the guidelines anyway, I just used the opportunity ; )

**Comment by b-g** [comment]
@dnlutz Wow! Great to have this list! Like especially the pattern approach of "problem and solution". Would be important to have this soonish in the styleguide.

**Comment by b-g** [comment]
Had finally time to create a hacker cat ... changed the size of the head, the clothing to the one of the mechanic (less complex than the one of the technologist) and added a "logo" to the notebook. Will go with the last one and create a PR soon. Hope that is fine for everyone.

<img width="855" alt="Screen Shot 2020-03-06 at 11 56 55" src="https://user-images.githubusercontent.com/480224/76077948-f4397100-5fa1-11ea-9aba-722e7a9fbac1.png">

**Comment by b-g** [comment]
Hacker cat will be shipped with next release ...

**Comment by alexkadis** [comment]
Thank you! I'm sorry that I wasn't able to complete it.

**Comment by b-g** [comment]
@alexkadis No worries! :) Thanks for the push/suggestion!

**Comment by alexkadis** [comment]
@b-g I would like to be listed as the author of this emoji. I didn't complete it but I did do most of the work. Is that ok? I've just always wanted to have 'my own' emoji.

**Comment by b-g** [comment]
Hi @alexkadis,

Sorry for the slow response very busy here! 

I'm afraid but I see this very different. You are absolutely right that you put work in it and that you've proposed it. However you did not push it through and you did not communicate with us for 3 months. Hence I felt safe that you would not be any longer interested and took over the todo. 

I can assure you that I did not incorporate anything new from your side, not any svg source code nor new ideas. I very literary took the already existing cat (scaled it down), added a trouser (based on the already existing mechanic) and added a notebook (3d not 2d, based on the already existing technologist).

But we gave you credits for the suggestion, see https://github.com/hfg-gmuend/openmoji/blob/master/changelog.txt#L190

I hope this is okay for your.

> I've just always wanted to have 'my own' emoji.

That is great! What about designing a new one e.g. still a few open for Emoji 13? Or there a some very cool non standard ones in a similar realm like the hack cat?

<img width="182" alt="Screen Shot 2020-07-13 at 14 28 15" src="https://user-images.githubusercontent.com/480224/87304583-54728d80-c515-11ea-9f5b-5dd09fa8388f.png">


We really would love to see your contribution here!

All the best,
Benedikt
